src/commands/channels/create.ts

```diff
@@ -0,0 +1,145 @@
+import {MixClient, MixResponse, MixResult} from '../../mix/types'
+import MixCommand from '../../utils/base/mix-command'
+import * as MixFlags from '../../utils/flags'
+import {DomainOption, validateChannelColor, validateChannelModeOptions} from '../../utils/validations'
+
+const debug = makeDebug('mix:commands:channels:create')
+
+export default class ChannelsCreate extends MixCommand {
+  static description = `create a new channel
+
+Use this command to create a new channel in your Mix project.
+A channel defines a collection of different modalities through
+which users interact with your application. The currently available
+modalities are:
+
+audioscript
+dtmf
+interactivity
+richtext
+tts
+
+Note that, for your convenience, this command will automatically convert
+inputs to the --mode flag as close to the format of the above modes as possible,
+by converting upper-case letters to lower-case and removing deliminating punctuation
+('-' and '_'). So, the modes 'audioscript', 'AUDIO-SCRIPT', and 'a_u-d_i-o----scriPT'
+are all equivalent to the server-side value of 'AUDIO_SCRIPT'. 
+
+The same conversion is done for colours, except that dashes ('-') are replaced 
+with ('_') rather than being deleted. So, 'light-pink', 'light____pink', and 
+'LIGHT_PINK' are also equivalent.
+
+Acceptable channel colors are:
+
+blue          brown         corn-flower
+cyan          green         grey
+indigo        light-green   light-grey  
+light-orange  light-pink    light-purple
+orange        pink          purple
+ruby          salmon        sky
+teal          yellow
+
+${chalk.bold('IMPORTANT:')} Due to a current server-side limitation,
+the command currently requires that both the 
+--mode and --color flags are set.`
+
+  static examples = [
+    `mix channels:create -P 1922 --name "New IVR channel" \\
+    --mode tts --mode interactivity --color light-pink`,
+  ]
+
+  static flags = {
+    color: MixFlags.required(MixFlags.channelColorFlag),
+    mode: MixFlags.modesFlag,
+    name: flags.string({
+      description: 'channel name',
+      required: true,
+    }),
+    project: MixFlags.projectWithDefaultFlag,
+    // output flags
+    json: MixFlags.jsonFlag,
+    yaml: MixFlags.yamlFlag,
+  }
+
+  get domainOptions(): DomainOption[] {
+    debug('get domainOptions()')
+    return ['project']
+  }
+
+  tryDomainOptionsValidation(options: any, domainOptions: DomainOption[]) {
+    debug('tryDomainOptionsValidation()')
+    super.tryDomainOptionsValidation(options, domainOptions)
+
+    if (options.color !== undefined) {
+      validateChannelColor(options.color)
+    }
+
+    if (options.mode !== undefined) {
+      validateChannelModeOptions(options.mode)
+    }
+  }
+
+  async buildRequestParameters(options: Partial<flags.Output>): Promise<ChannelsCreateParams> {
+    debug('buildRequestParameters()')
+
+    const {
+      project: projectId,
+      name: displayName,
```

Comment:
I was able to create a channel with --name "    MyNewChannel   " and the channel got created with the left and right space paddings. Perhaps we could trim() `name`?

Reply:
It also accepted --name "" (empty string). We should disallow empty string.